keps/sig-storage/2268-non-graceful-shutdown/README.md

```diff
@@ -0,0 +1,466 @@
+  <!--
+  Any change of default behavior may be surprising to users or break existing
+  automations, so be extremely careful here.
+  -->
+  Yes. If this feature is enabled, the pod gc controller in kube-controller-manager
+  would need to ensure that `SafeDetach` option in `CSIDriver` is true before
+  forcefully deleting pods.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+  Describe the consequences on existing workloads (e.g., if this is a runtime
+  feature, can it break the existing applications?).
+  Yes. If this feature is disable once it has been enabled, it falls back to
+  default behavior before the feature is enabled, meaning the pod gc controller
+  will not check the `SafeDetach` option in `CSIDriver` and make decision
+  depending on that.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  If we reenable the feature if it was previously rolled back, the pod gc controller
+  will again check the `SafeDetach` option in `CSIDriver` and make sure it is
+  true before forcefully deleting pods.
+
+* **Are there any tests for feature enablement/disablement?**
+  <!--
+  The e2e framework does not currently support enabling or disabling feature
+  gates. However, unit tests in each component dealing with managing data, created
+  with and without the feature, are necessary. At the very least, think about
+  conversion tests if API types are being modified.
+  -->
+  We will add unit test for feature enablement/disablement.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g., what if some components will restart
+   mid-rollout?
+
+* **What specific metrics should inform a rollback?**
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and can't do that now.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+  Even if applying deprecation policies, they may still surprise some users.
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+  checking if there are objects with field X set) may be a last resort. Avoid
+  logs or events for this purpose.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [ ] Metrics
```

Comment:
A reminder that you'll need to have metrics for beta.

Reply:
Sure.